Armadillo/src/main/java/com/scribd/armadillo/models/Models.kt

```diff
@@ -143,6 +144,36 @@ data class Chapter(
 
 data class DrmInfo(val drmType: DrmType, val licenseServer: String, val drmHeaders: Map<String, String> = emptyMap()) : Serializable
 
+@kotlinx.serialization.Serializable
+data class DrmDownload(
+    val drmKeyId: ByteArray,
+    val drmType: DrmType,
+    val licenseServer: String,
+    @ContentType val audioType: Int,
+) {
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as DrmDownload
+
+        if (!drmKeyId.contentEquals(other.drmKeyId)) return false
+        if (drmType != other.drmType) return false
+        if (licenseServer != other.licenseServer) return false
+        if (audioType != other.audioType) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = drmKeyId.contentHashCode()
+        result = 31 * result + drmType.hashCode()
+        result = 31 * result + licenseServer.hashCode()
+        result = 31 * result + audioType
+        return result
+    }
```

Comment:
We should get this by default with data class, why do we need to do it by hand?

Reply:
It's because this class contains a field of type ByteArray. Not generating equals and hashCode results in a warning